File: Leetcode Solutions/2 Weeks Study Plan to Tackle DS/DS 14-Day Challenge/Day-11/Maximum Depth Of Binary Tree.java
public int maxDepth(TreeNode root) {
    if(root==null)
        return 0;
 
    int leftDepth = maxDepth(root.left);
    int rightDepth = maxDepth(root.right);
 
    int bigger = Math.max(leftDepth, rightDepth);
 
    return bigger+1;
}

File: 01. DataStructures/04. Tree/Leetcode Questions/maximum_depth_of_binary_tree.js


var maxDepth = function(root) {
   
   if(!root) return 0;
   let current = 0;
   let max = 0;

    function dfs(root, current) {
        if(!root) return;
        
        if(!root.left && !root.right) {
            max = Math.max(current, max);
            return;
        }

        dfs(root.left, current + 1);
        dfs(root.right, current + 1);

    }

   dfs(root, current + 1);
   return max;
};
